File: aat-android/src/main/java/ch/bailu/aat/activities/NominatimActivity.java
package ch.bailu.aat.activities;

import ch.bailu.aat.util.NominatimApi;
import ch.bailu.aat_lib.search.poi.OsmApiConfiguration;
import ch.bailu.aat.views.bar.MainControlBar;
import ch.bailu.aat_lib.coordinates.BoundingBoxE6;

public class NominatimActivity extends AbsOsmApiActivity {

    @Override
    public OsmApiConfiguration createApiConfiguration(BoundingBoxE6 boundingBox)  {
        return new NominatimApi(this, boundingBox) {
            @Override
            protected String getQueryString() {
                return editorView.toString();
            }
        };
    }

    @Override
    public void addCustomButtons(MainControlBar bar) {
        bar.addSpace();
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/activities/NominatimActivity.kt
package ch.bailu.aat.activities

import ch.bailu.aat_lib.api.nominatim.NominatimApi
import ch.bailu.aat.views.bar.MainControlBar
import ch.bailu.aat_lib.api.ApiConfiguration

class NominatimActivity : AbsOsmApiActivity() {

    public override fun createApiConfiguration(): ApiConfiguration {
        return object : NominatimApi(appContext) {
            override val queryString: String
                get() = editorView.toString()
        }
    }

    public override fun addCustomButtons(bar: MainControlBar) {
        bar.addSpace()
    }
}
